api/listings.py

```diff
@@ -91,6 +91,25 @@ def by_area(self, bbox, limit=None, **params):
 
     @cherrypy.expose
     @cherrypy.tools.json_out()
+    def by_ids(self, ids, limit=None, **params):
+        i = ids.split(',')
+        limit = self.num_param(limit, 20, 100)
+
+        mapdb = cherrypy.request.app.config['DB']['map']
+        r = mapdb.tables.routes.data
+        s = mapdb.tables.segments.data
+        h = mapdb.tables.hierarchy.data
+
+        res = sa.select([r.c.id, r.c.name, r.c.intnames, r.c.symbol, r.c.level])\
+               .where(r.c.id.in_(i))\
+               .order_by(r.c.level, r.c.name)\
+               .limit(limit)
+
+        return self.create_list_output('ids', ",".join(i),
```

Comment:
Just 'i' for second parameter. It will then be correctly rendered as a json array.